reconcile/aus/ocm_addons_upgrade_scheduler_org.py

```diff
@@ -0,0 +1,158 @@
+                ocm_map=ocm_map,
+                addon_id=addon_state.addon_id,
+                integration=self.name,
+            )
+            diffs = calculate_diff(
+                addon_state.current_state,
+                addon_state.desired_state,
+                ocm_map,
+                version_data_map,
+                addon_id=addon_state.addon_id,
+            )
+            aus.act(dry_run, diffs, ocm_map, addon_id=addon_state.addon_id)
+
+    def get_organization_upgrade_spec(
+        self, org_name: Optional[str] = None
+    ) -> dict[str, OrganizationUpgradeSpec]:
+        return organization_upgrade_spec(org_name=org_name)
+
+
+def get_state_for_org_spec(
+    org_upgrade_spec: OrganizationUpgradeSpec, fetch_current_state: bool
+) -> tuple[OCMMap, list[dict[str, Any]], list[dict[str, Any]]]:
+    settings = queries.get_app_interface_settings()
+    cluster_like_objects = [
+        policy.dict(by_alias=True) for policy in org_upgrade_spec.specs
+    ]
+    ocm_map = OCMMap(
+        clusters=cluster_like_objects,
+        integration=QONTRACT_INTEGRATION,
+        settings=settings,
+        init_version_gates=True,
+        init_addons=True,
+    )
+    desired_state = aus.fetch_desired_state(cluster_like_objects, ocm_map, addons=True)
+    current_state: list[dict[str, Any]] = []
+    if fetch_current_state:
+        current_state = aus.fetch_current_state(
+            cluster_like_objects,
+            ocm_map,
+            addons=True,
+        )
+
+    return (ocm_map, current_state, desired_state)
+
+
+def get_state_for_org_spec_per_addon(
+    org_upgrade_spec: OrganizationUpgradeSpec, fetch_current_state: bool
+) -> tuple[OCMMap, list[AUSOrgAddonUpgradeState]]:
+    ocm_map, current_state, desired_state = get_state_for_org_spec(
+        org_upgrade_spec, fetch_current_state
+    )
+    result = []
+    for addon_id in set(a["addon_id"] for a in desired_state):
+        addon_current_state = [c for c in current_state if c["addon_id"] == addon_id]
+        addon_desired_state = [d for d in desired_state if d["addon_id"] == addon_id]
+        result.append(
+            AUSOrgAddonUpgradeState(
+                addon_id=addon_id,
+                current_state=addon_current_state,
+                desired_state=addon_desired_state,
+            )
+        )
+    return ocm_map, result
+
+
+def organization_upgrade_spec(
+    org_name: Optional[str] = None,
+) -> dict[str, OrganizationUpgradeSpec]:
+    return {
+        org.name: OrganizationUpgradeSpec(
+            org=org,
+            specs=[
+                ClusterUpgradeSpec(
+                    name=cluster.name,
+                    ocm=org,
+                    upgradePolicy=cluster.upgrade_policy,
+                )
+                for cluster in org.upgrade_policy_clusters or []
+            ],
+        )
+        for org in aus_organizations_query(query_func=gql.get_api().query).organizations
+        or []
```

Comment:
Is there a risk that `aus_organizations_query(query_func=gql.get_api().query)` is `None` or `{}` here?

Reply:
this is a type hint problem. the definition of almost all entries in the GQL schema `Query` lack an `isRequired=true` and therefore qenerate has to assume that it can potentially be None. in reality all entries of this sort will always return a list. it might be empty but it is a list.